protocol/storage/fault_slasher.go

```diff
@@ -92,6 +97,18 @@ func (sfm *FaultSlasher) Slash(ctx context.Context, currentHeight *types.BlockHe
 
 	// Slash late miners.
 	for minerStr, state := range *lms {
+		if _, ok := sfm.slashed[minerStr]; ok {
+			// Skip slashed miner.
+			// This logic is not perfect. A miner that appears to be late could redeem itself
+			// (e.g. due to a chain re-org) such that a submitted slashing message fails, but
+			// this slasher will remember that it has attempted to slash that miner, and won't
+			// re-attempt to do so in the future.
+			// The `slashed` cache key should include the proving period for which the attempt was
+			// made so that if a miner does enter another period, it becomes slashable again.
+			// Getting that information requires painful plumbing of data through actor methods
```

Comment:
This actually only requires including MessageQuery, calling the `getProvingPeriod` method on the miner actor, and parsing the result. Still possibly overkill given the likelihood of a problem.